api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -426,16 +428,40 @@ protected Type.TypeID typeId() {
   }
 
   static class TimestampLiteral extends ComparableLiteral<Long> {
-    TimestampLiteral(Long value) {
+    private final ChronoUnit unit;
+
+    TimestampLiteral(ChronoUnit unit, Long value) {
       super(value);
+      this.unit = unit;
     }
 
     @Override
     @SuppressWarnings("unchecked")
     public <T> Literal<T> to(Type type) {
       switch (type.typeId()) {
         case TIMESTAMP:
-          return (Literal<T>) this;
+          ChronoUnit toUnit = ((TimestampType) type).unit();
+          switch (unit) {
+            case MICROS:
```

Comment:
I thought that micros was not a supported unit. Why would `TimestampType` use it?